docs/markdown/Python/python-goals/python-run-goal.md

```diff
@@ -36,6 +42,26 @@ The program will have access to the same environment used by the parent `./pants
 >
 > Run `./pants dependencies --transitive path/to/binary.py` to ensure that all the files you need are showing up, including for any [assets](doc:assets) you intend to use.
 
+Execution Semantics
+-------------------
+
+Running a `pex_binary` is equivalent to `package`-ing the target followed by executing the built PEX
+from the repo root.
+
+Running a `python_source` with the `run_goal_use_sandbox` field set to `True` (the default) runs your
+code in an ephemeral sandbox (sometimes referred to as the "chroot") with your firstparty code and
+Pants-generated files (such as a `relocated_files` or `archive`) copied inside.
+
+Running a `python_source` with the `run_goal_use_sandbox` field set to `False` is equivalent to
+running the source directly (a la `python ...`) with the set of third-party dependencies exposed to
+the interpreter. This is comparable to using a virtual environment or Poetry to run your script
+(E.g. `venv/bin/python ...` or `poetry run python ...`).
+
+The main tradeoff between the two values is if you need to load Pants-generated files
+(set the field to `True`) or want to write to in-repo files such as using Django's
+`manage.py makemigrations` (set the field to `False`).
```

Comment:
Maybe inline this paragraph into the two previous ones:

> If you are using generated files like this, you may need to set the `run_goal_use_sandbox` to `True` for file loading to work properly.

> When scripts write in-repo files—such as Django's `manage.py makemigrations`—it is often necessary to set `run_goal_use_sandbox` to `False` so that the file is written into the expected location.